src/App.tsx

```diff
@@ -1,24 +1,196 @@
+
+          return item;
+        });
+      });
+    } catch (error) {
+      setErrorMessage(ErrorMessages.OnUpdate);
+    }
+
+    setLoadingTodoIds(prevIds => prevIds.filter(prevId => prevId !== todoId));
+  }, []);
+
+  const updateTitle = useCallback(async (
+    todoId: number,
+    title: string,
+  ) => {
+    setLoadingTodoIds(prevIds => ([...prevIds, todoId]));
+    try {
+      await updateTodoTitle(todoId, title);
+      setErrorMessage(ErrorMessages.NoError);
+
+      setTodos((prevTodos: Todo[]): Todo[] => {
+        return prevTodos.map((item: Todo) => {
+          if (item.id === todoId) {
+            item.title = title; // eslint-disable-line no-param-reassign
+          }
+
+          return item;
+        });
+      });
+    } catch (error) {
+      setErrorMessage(ErrorMessages.OnUpdate);
+    }
+
+    setLoadingTodoIds(prevIds => prevIds.filter(prevId => prevId !== todoId));
+  }, []);
+
+  const toggleAll = useCallback(() => {
+    let newTodoStatus = true;
+    let todosToUpdate = todos;
+
+    if (isAllTodosCompleted) {
+      newTodoStatus = false;
+    } else if (isSomeTodosCompleted) {
+      newTodoStatus = true;
+      todosToUpdate = getFilteredTodos(todos, FilterOptions.Active);
+    }
+
+    todosToUpdate.forEach(todo => handleTogglingTodo(todo.id, newTodoStatus));
+  }, [todos]);
+
+  const loadTodos = async () => {
+    try {
+      setErrorMessage(ErrorMessages.NoError);
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorMessage(ErrorMessages.OnLoad);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          setTempTodo={setTempTodo}
+          tempTodo={tempTodo}
+          userId={USER_ID}
+          setErrorMessage={setErrorMessage}
+          setTodos={setTodos}
+          isAllTodosCompleted={isAllTodosCompleted}
+          toggleAll={toggleAll}
+        />
+        {!!todos.length && (
+          <TodoList
+            todos={visibleTodos}
+            handleDeleteTodo={handleDeleteTodo}
+            loadingTodoIds={loadingTodoIds}
+            handleTogglingTodo={handleTogglingTodo}
+            updateTitle={updateTitle}
+          />
+        )}
+        {tempTodo && (
+          <TodoItem
+            todo={tempTodo}
+            loadingTodoIds={loadingTodoIds}
+          />
+        )}
+
+        {!!todos.length && (
```

Comment:
footer should also be visible when there is a tempTodo

Reply:
agree